src/Repository/ProductCommentCriterionRepository.php

```diff
@@ -88,4 +253,116 @@ public function getByProduct($idProduct, $idLang)
 
         return $qb->execute()->fetchAll();
     }
+
+    /**
+     * Get Criterions
+     *
+     * @return array Criterions
+     */
+    public function getCriterions($id_lang, $type = false, $active = false)
+    {
+        $sql = '
+			SELECT pcc.`id_product_comment_criterion`, pcc.id_product_comment_criterion_type, pccl.`name`, pcc.active
+			FROM `' . _DB_PREFIX_ . 'product_comment_criterion` pcc
+			JOIN `' . _DB_PREFIX_ . 'product_comment_criterion_lang` pccl ON (pcc.id_product_comment_criterion = pccl.id_product_comment_criterion)
+			WHERE pccl.`id_lang` = ' . $id_lang . ($active ? ' AND active = 1' : '') . ($type ? ' AND id_product_comment_criterion_type = ' . (int) $type : '') . '
+			ORDER BY pccl.`name` ASC';
+        $criterions = $this->connection->executeQuery($sql)->fetchAll();
+
+        $types = self::getTypes();
+        foreach ($criterions as $key => $data) {
+            $criterions[$key]['type_name'] = $types[$data['id_product_comment_criterion_type']];
+        }
+
+        return $criterions;
+    }
+
+    /**
+     * @param int $id_criterion
+     *
+     * @return array
+     */
+    public function getProducts(int $id_criterion)
+    {
+        $sql = '
+			SELECT pccp.id_product, pccp.id_product_comment_criterion
+			FROM `' . _DB_PREFIX_ . 'product_comment_criterion_product` pccp
+			WHERE pccp.id_product_comment_criterion = ' . $id_criterion;
+
+        $res = $this->connection->executeQuery($sql)->fetchAll();
+
+        $products = [];
+        if ($res) {
+            foreach ($res as $row) {
+                $products[] = (int) $row['id_product'];
+            }
+        }
+
+        return $products;
+    }
+
+    /**
+     * @param int $id_criterion
+     *
+     * @return array
+     */
+    public function getCategories(int $id_criterion)
+    {
+        $sql = '
+			SELECT pccc.id_category, pccc.id_product_comment_criterion
+			FROM `' . _DB_PREFIX_ . 'product_comment_criterion_category` pccc
+			WHERE pccc.id_product_comment_criterion = ' . $id_criterion;
+
+        $res = $this->connection->executeQuery($sql)->fetchAll();
+
+        $criterions = [];
+        if ($res) {
+            foreach ($res as $row) {
+                $criterions[] = (int) $row['id_category'];
+            }
+        }
+
+        return $criterions;
+    }
+
+    /**
+     * @return array
+     */
+    public function getTypes()
+    {
+        $sfTranslator = SymfonyContainer::getInstance()->get('translator');
```

Comment:
You could inject translator service in the construtor

Reply:
I tried inserting `@translator` in common.yml but got this error
> Fatal error: Uncaught Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The service "product_comment_criterion_repository" has a dependency on a non-existent service "translator".

Then I discovered that ps_linklist uses `@translator` in [its services](https://github.com/PrestaShop/ps_linklist/blob/dev/config/services.yml), `Symfony\Contracts\Translation\TranslatorInterface` in [its repository](https://github.com/PrestaShop/ps_linklist/blob/dev/src/Repository/LinkBlockRepository.php#L31), and dropped support for both PS 1.7.8, PS 8.0 :cry:

Reply:
With the upgrade to symfony 4, `Symfony\Component\Translation\TranslatorInterface` is transferred to `Symfony\Contracts\Translation\TranslatorInterface`. On the other hand, the translator service is still present, it has just changed the package.

Reply:
Injecting translator service create error so I'd prefer do it in a minor version.